nomad/structs/structs.go

```diff
@@ -2234,6 +2235,10 @@ func (r *Resources) Validate() error {
 		}
 	}
 
+	if r.MemoryMaxMB != 0 && r.MemoryMaxMB < r.MemoryMB {
+		mErr.Errors = append(mErr.Errors, fmt.Errorf("MemoryMaxMB value (%d) should be larger than MemoryMB value (%d)", r.MemoryMaxMB, r.MemoryMB))
+	}
```

Comment:
The only validation we do for MemoryMaxMB is that it needs to be equal or higher than MemoryMB. The scheduler may place the alloc on a client with less memory than MemoryMaxMB and the client may run it.

It's unclear what the behavior should be: ideally, we prioritize placing the job allocations on clients that exceed the MemoryMaxMB, but they should be run IMO even if the only nodes available are nodes that meet the basic memory requirements but not the max ones. Also, I suspect some operators will set high values as an optimistic "just be lenient and give me some access memory" rather than setting max values through rigorous analysis and experimentation.

Reply:
This is an interesting question - when clusters start to fill up, there's an incentive to set Memory absurdly low to increase your chance of getting scheduled, then lean on MemoryMax for resources to actually run. Should MemoryMax at least feed into Quota?